### DISCUSSION TITLE
Benetel RAN550 firmware v1.0.4

### DISCUSSION THREAD

**Discussion by khooi8913** [srsran/srsRAN_Project#487] [General Help] [answered]
It was known and stated in the documentation previously that srsRAN was integrated with the v0.8.1_patch2 firmware for the Benetel RAN550. How about the more recent v1.0.4 firmware? Has anyone tested it?

**Comment by andrepuschmann** [comment, marked as answer]
Hey,
yes, we are currently working on the 1.0.4 integration. Tests are still ongoing and as soon as they are completed we will of course update the RU integration section in the docs.
Meanwhile if you want to give it a try yourself have a look at the attached files. The RU config is pretty much untouched, except for the FlexRAN workaround which we don't required. So please disable this. Main difference in the gNB config is that we use the short PRACH format now (PRACH config section) and that we disable the RU bandwidth "hack". The RU supports now lower bandwidths, e.g. 20 and 40 MHz, not just 100 MHz. Also we enable OFH CPlane request for PRACH (`is_prach_cp_enabled=true'). The attached config is for 20MHz using DPDK. Make sure to have the RU configured for this bandwidth and also the carrier freqeuency to 3.6GHz.

Hope that helps.

* [ru_config.txt](https://github.com/srsran/srsRAN_Project/files/14396193/ru_config.txt) -- (compare with file on RU and adjust values) 
* [gnb_benetel_new_firmware.txt](https://github.com/srsran/srsRAN_Project/files/14396200/gnb_benetel_new_firmware.txt) -- (adjust and rename to yml)

**Reply by IbrahimBabiker87** [reply]
@andrepuschmann ,

The above configuration working fine for bandwidth 20,40 Mhz, however for 80 and 100 MHz, The UE connect only occasionally " UE will not always connect" the cell is detected but no RACH is detected.

Do you observe the same issue ? Do you already tested with 80 & 100 Mhz ? is there any special configurations ?

```
2024-02-29T12:01:21.251520 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '959.9' and sector#0
2024-02-29T12:01:21.256023 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '959.18' and sector#0
2024-02-29T12:01:21.256520 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '959.19' and sector#0
2024-02-29T12:01:21.271018 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '961.8' and sector#0
2024-02-29T12:01:21.271531 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '961.9' and sector#0
2024-02-29T12:01:21.276022 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '961.18' and sector#0
2024-02-29T12:01:21.276521 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '961.19' and sector#0
2024-02-29T12:01:21.291013 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '963.8' and sector#0
2024-02-29T12:01:21.291519 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '963.9' and sector#0
2024-02-29T12:01:21.296020 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '963.18' and sector#0
2024-02-29T12:01:21.296526 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '963.19' and sector#0
2024-02-29T12:01:21.311027 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '965.8' and sector#0
```

**Reply by andrepuschmann** [reply]
Testing was done with 20 and 40 MHz only so far due to time constraints. We plan to expand the testing in the following days.

**Comment by khooi8913** [comment]
Hi Andre, thank you for the response, it is reassuring.
It very much fits with our understanding and expectations for the new release, and this makes us more confident in making the upgrade first.
We have also been using some of the "hacks", it is good to know that we no longer need them.

**Comment by andrepuschmann** [comment]
I am reopening the discussion as integration and testing is still ongoing and this could be useful to others.

**Reply by khooi8913** [reply]
As feedback, we upgraded to v1.0.4 and can confirm it works in our setup at 40MHz for 4x2.
In the shared configs, it did not work for us though. 
Instead, we modified to "static_compressed", and dl_broadcast_flag to "false".

**Reply by andrepuschmann** [reply]
The `static_compressed` setting should not be required if both configs match. It seems they don't in your case. The `dl_broadcast_flag=true` is  something that we suggest for now because of an issue of the RU firmware when only using one antenna port. If you connect two ports it should work with `false` as well. We are working with Benetel on it.

**Comment by AzeezEbrahim** [comment]
hey @khooi8913, i want to ask about the error loss rate, did you manage to reduce it ? another thing, are you using DPDK ?
thanks in advance

**Reply by khooi8913** [reply]
Hi @AzeezEbrahim , we upgraded the RU to v1.0.4, and from what I could tell, for 40MHz, the error rates are much better.
As for DPDK, no, we have not tried with DPDK.

**Comment by aibtw** [comment]
As feedback, we can also confirm that RAN550 v1.0.4 works with the exact conf shared by @andrepuschmann above, except is_dl_broadcast_enabled is set to false and using 4 tx instead of 2. 

We tried 20 MHz and 40 MHz, 4x2 and 4x4. All working.
100 MHz is not working, but I can see in the logs that `[OFH     ] [W] Ethernet transmitter call to sendto failed as it could not transmit '7680' bytes, consider tuning the NIC system settings to obtain higher performance or use DPDK` so maybe that's the issue. 

However, there is still high loss rate on both uplink and downlink in all the configurations we tried. Would be really grateful if @khooi8913 you can share your du-ru configs in case the loss got better, maybe there is something that I am missing.

**Reply by VikrantDigiCat** [reply]
Trying to get a srsRAN gNodeB to work with a Foxconn Indoor RU but getting below error on the gNodeB. Can anyone help to resolve this?
_2025-05-27T17:15:01.734935 [OFH     ] [W] sendto failed to transmit 7680 bytes, consider tuning the NIC for higher performance_

**Reply by aibtw** [reply]
did you run the srsRAN performance script? Looks like MTU Size issue.

**Reply by VikrantDigiCat** [reply]
@aibtw Thanks for the suggestion. Issue got resolved after enabling DPDK.

**Comment by ffxxcad** [comment]
Hello @khooi8913 and @aibtw, which UEs have you been able to attach?

**Reply by aibtw** [reply]
So far, any UE that worked with USRP before, worked with us now. 
examples are the OnePlus Nord N30 5G, and the realme 9i 5G.

**Comment by OsmanMertYilmaz** [comment]
Hi, hope you're doing well. We are also experiencing a similar issue, as shown below. Currently, the COTS UE is unable to detect the network. Do you think this issue is entirely related to DPDK?

`[OFH     ] [W] Ethernet transmitter with fd = '8' could not transmit '7680' bytes, consider tuning the NIC system settings to obtain higher performance or use DPDK`
